src/utils.ts

```diff
@@ -126,6 +124,58 @@ export function equal(target: ObjectInterface, base: ObjectInterface): boolean {
 	return every(target, (v, k) => v === base[k]);
 }
 
-export function toFixed(num: number) {
-	return Math.round(num * FIXED_DIGIT) / FIXED_DIGIT;
+let lastRoundNum = -1;
+let roundNumFunc = null;
+
+export function roundNumber(num: number, roundUnit: number) {
+	// Cache for performance
+	if (!roundNumFunc || lastRoundNum !== roundUnit) {
+		roundNumFunc = getRoundFunc(roundUnit);
+		lastRoundNum = roundUnit;
+	}
+
+	return roundNumFunc(num);
+}
+
+export function roundNumbers(num: ObjectInterface<number>, roundUnit: ObjectInterface<number> | number) {
+	if (!num || !roundUnit) {
+		return num;
+	}
+	const isNumber = typeof roundUnit === "number";
+	return map(num, (value, key) => roundNumber(value, isNumber ? roundUnit : roundUnit[key]));
+}
+
+// https://jsperf.com/precision-calculation
+export function getDecimalPlace(val: number): number {
+	if (!isFinite(val)) {
+		return 0;
+	}
+
+	let p = 0;
+	let e = 1;
+
+	while (Math.round(val * e) / e !== val) {
+		e *= 10;
+		p++;
+	}
+
+	return p;
+}
+
+export function reversePow(n: number) {
+	// replace Math.pow(10, -n) to solve floating point issue.
+	// eg. Math.pow(10, -4) => 0.00009999999999999999
+	return 1 / Math.pow(10, n);
+}
+
+export function getRoundFunc(v: number) {
```

Comment:
I think it's essential is not where it is used, but it caches expensive function calls.

Reply:
Is it meaningful to cache a one-time function?

Reply:
Is it meaningful to cache a function that is only last used and is easy to change?

Reply:
I thought it can be more common that the user uses the same `rounUnit` for different values.
In that case, I think it can be meaningful than not doing it. (it might be still trivial though)